proposals/12147-mlmd-removal/schema_changes.sql

```diff
@@ -0,0 +1,100 @@
+CREATE TABLE `artifacts`
+(
+    `UUID`            varchar(191) NOT NULL,
+    `Namespace`       varchar(63)  NOT NULL,              -- enables multi-tenancy on artifacts
+    `Type`            varchar(64)           DEFAULT NULL, -- examples: Artifact, Model, Dataset
+    `Uri`             text,
+    `Name`            varchar(128)          DEFAULT NULL,
+    `CreatedAtInSec`  bigint       NOT NULL DEFAULT '0',
+    `LastUpdateInSec` bigint       NOT NULL DEFAULT '0',
+    `Properties`      JSON                  DEFAULT NULL, -- equivalent to mlmd custom properties
+    PRIMARY KEY (`UUID`),
+    KEY               idx_type_namespace (`Namespace`, `Type`),
+    KEY               idx_created_timestamp (`CreatedAtInSec`),
+    KEY               idx_last_update_timestamp (`LastUpdateInSec`)
+);
+
+-- Analogous to an mlmd Event, except it is specific to artifacts <-> tasks (instead of executions)
+CREATE TABLE `artifact_events`
+(
+    `UUID`           varchar(191) NOT NULL,
+    `ArtifactID`     varchar(191) NOT NULL,
+    `TaskID`         varchar(191) NOT NULL,
+    -- 0 for INPUT, 1 for OUTPUT
+    `Type`           int          NOT NULL,
+    `CreatedAtInSec` bigint       NOT NULL DEFAULT '0',
+
+    PRIMARY KEY (`UUID`),
+    UNIQUE KEY `UniqueLink` (`ArtifactID`,`TaskID`,`Type`),
+    KEY              `idx_link_task_id` (`TaskID`),
+    KEY              `idx_link_artifact_id` (`ArtifactID`),
+    KEY              `idx_created_timestamp` (`CreatedAtInSec`),
+    CONSTRAINT fk_artifact_events_tasks FOREIGN KEY (TaskID) REFERENCES tasks (UUID) ON DELETE CASCADE ON UPDATE CASCADE,
+    CONSTRAINT fk_artifact_events_artifacts FOREIGN KEY (ArtifactID) REFERENCES artifacts (UUID) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+CREATE TABLE `tasks`
+(
+    `UUID`             varchar(191) NOT NULL,
+    `Namespace`        varchar(63)  NOT NULL, -- updated to 63 (max namespace size in k8s)
+    -- This is used for searching for cached_fingerprints today
+    -- likely to prevent caching across pipelines 
+    `PipelineName`     varchar(128) NOT NULL,
+    `RunUUID`          varchar(191) NOT NULL,
+    `PodName`          varchar(255) NOT NULL, -- This is broken today and will need to be fixed
+    `CreatedAtInSec`   bigint       NOT NULL,
+    `StartedInSec`     bigint       DEFAULT '0',
+    `FinishedInSec`    bigint       DEFAULT '0',
+    `Fingerprint`      varchar(255) NOT NULL,
+    `Name`             varchar(128) DEFAULT NULL,
+    `ParentTaskUUID`   varchar(191) DEFAULT NULL,
+    `State`            varchar(64)  DEFAULT NULL,
+    `StateHistory`     longtext,
```

Comment:
I'm not sure I follow why this is text. It seems like it should be either JSON or the data is in a separate table that we join to.